Source/JavaScriptCore/runtime/JSBigInt.cpp

```diff
@@ -473,6 +473,39 @@ class Int32BigIntImpl {
     int32_t m_value;
 };
 
+class Int64BigIntImpl {
+public:
+    explicit Int64BigIntImpl(int64_t value)
+        : m_value(value)
+        , m_sign(value < 0)
+    {
+    }
+
+    explicit Int64BigIntImpl(uint64_t value)
+        : m_value(value)
+        , m_sign(false)
+    { }
+
+    ALWAYS_INLINE bool isZero() { return !m_value; }
+    ALWAYS_INLINE bool sign() { return m_sign; }
+    ALWAYS_INLINE unsigned length() { return isZero() ? 0 : 1; }
+    ALWAYS_INLINE JSBigInt::Digit digit(unsigned i)
+    {
+        ASSERT(length());
+        ASSERT_UNUSED(i, !i);
+        if (sign())
+            return static_cast<JSBigInt::Digit>(-static_cast<int64_t>(m_value));
+        return m_value;
+    }
+
+private:
+    friend struct JSBigInt::ImplResult;
+    JSBigInt::Digit m_value;
```

Comment:
Isn't this too small for 32-bit platforms since `JSBigInt::Digit = UCPURegister`?

We should probably just use uint64_t and fix `length()` and `digit`